### ISSUE TITLE
Panic when creating jet stream consumer on wildcard stream

### ISSUE THREAD

**Issue by areknoster** [nats-io/nats.go#1641]
### Observed behavior

When creating consumer against embedded NATS server JetStream with wildcard (`>`) subscription, the function panics.

### Expected behavior

The function doesn't panic and returns consumer that's usable.

### Server and client version

```
github.com/nats-io/nats-server/v2 v2.10.11
github.com/nats-io/nats.go v1.35.0
```

### Host environment

```
GOHOSTARCH='arm64'
GOHOSTOS='darwin'
GOVERSION='go1.22.3'
```

### Steps to reproduce

Run this test: #1640

**Comment by piotrpio** [comment]
Hello @areknoster, thanks for reporting the issue and thank you for creating the test - it looks like we're getting an unexpected response type from the server for create consumer request. I'll take a look and let you know when I have a fix.

**Comment by ripienaar** [comment]
The client should not allow a wildcard stream with ack set it will break everything.

**Comment by piotrpio** [comment]
The issue here is that if you create a stream with `>` in subjects, it means it will also be sending acks for all messages published on `>`, even if it's the JetStream API itself (so e.g. subject used to create the consumer). If you wish to have a catch all stream, it should be configured with `NoAck: true` option in stream config.

I'll be adding a validation in the client to disallow creating such streams.

**Comment by derekcollison** [comment]
@piotrpio I do not want us to disallow broader scoped subjects for streams but as you mention we should suppress pub acks if the subject is sufficiently wide. The server needs to support this of course and enforce the no pub ack config on those wider stream subjects.

**Comment by derekcollison** [comment]
@areknoster You are correct for certain subjects we should have the server perform the check.

**Comment by piotrpio** [comment]
@derekcollison Agreed, it would be good for the server to perform the validation.

@areknoster I agree that the client should not panic. We'll have to add a check for nil response in each JS API call. I will create an issue for that and if you wish to contribute that would be great!

**Comment by piotrpio** [comment]
@areknoster I've created the issue: https://github.com/nats-io/nats.go/issues/1642

If you wish to contribute, please assign yourself to it - thanks :)
